### PR TITLE
Random walker algorithm (resolves #2390)

### LINKED ISSUE #2390: Random walker algorithm

### PR DESCRIPTION
**Pull request by 0x00b1**

### CODE REVIEW

**Comment by paulmenzel** [comment]
Should the commits be squashed? The second one looks like a fix-up commit.

Also some notes about the implementation would be nice.

**Comment by 0x00b1** [comment]
@paulmenzel GitHub added a `Squash and merge` button! It rules! I also added some documentation. What do you think?

@mcquin This is ready for review!

**Comment by paulmenzel** [comment]
@0x00b1, I don’t like the *Squash and merge*, as there are often good reasons, for having a branch consist of more commits, to make it better digestible.

I also don’t know, if it will work, when you have merge the master branch back into your branch.

Also merging the branches, makes it hard to review now. At least in the file changes, I see `scikit-image` added to the depends. Or is that from one of your commits?

**Comment by 0x00b1** [comment]
random_walker_algorithm__6__image_cycle__1

**Comment by 0x00b1** [comment]
> @paulmenzel Also merging the branches, makes it hard to review now. At least in the file changes, I see scikit-image added to the depends. Or is that from one of your commits?

This was a mistake. 😳

**Comment by 0x00b1** [comment]
> @paulmenzel I don’t like the Squash and merge, as there are often good reasons, for having a branch consist of more commits, to make it better digestible.

I agree. However, I think in this case one commit makes sense.

**Comment by paulmenzel** [comment]
Agreed. In this case it probably does.

File: cellprofiler/modules/randomwalkeralgorithm.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import cellprofiler.module
+import cellprofiler.object
+import cellprofiler.setting
+import numpy
+import skimage.measure
+import skimage.segmentation
+
+
+class RandomWalkerAlgorithm(cellprofiler.module.ImageSegmentation):
+    module_name = "Random walker algorithm"
```
Comment (mcquin) [inline cellprofiler/modules/randomwalkeralgorithm.py]: For visual consistency, this should be "RandomWalkerAlgorithm".

File: cellprofiler/modules/randomwalkeralgorithm.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# coding=utf-8
+
+"""
+
+Random walker algorithm
+
+Single-channel images can be two-or-three-dimensional and multichannel images can be three-or-four-dimensional.
```
Comment (mcquin) [inline cellprofiler/modules/randomwalkeralgorithm.py]: I think it's nice to provide a description of what this module does.

**Pull request merged**